vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java

```diff
@@ -60,6 +68,48 @@ public class Popover extends Component implements HasComponents {
      */
     public Popover() {
         getElement().getNode().addAttachListener(this::attachComponentRenderer);
+
+        setTrigger(Set.of(PopoverTrigger.CLICK));
+    }
+
+    /**
+     * Sets the popover trigger mode describing how the overlay is opened or
+     * closed on user interaction with the target.
+     *
+     * When passing null or an empty set, the popover can only be opened and
+     * closed programmatically.
+     *
+     * @param trigger
+     *            the set of trigger values to set
+     */
+    public void setTrigger(Set<PopoverTrigger> trigger) {
```

Comment:
Just curious if we have any other API that allows to set multiple enum values? This way of doing it is fine IMO, just checking if there is an existing approach maybe. There's theme variants, but the methods for those are for adding / removing, rather than replacing.

An alternative could be to use variable arguments, which would not require to construct a set. No need to change this immediately, just opening a discussion.

Reply:
I'm also not sure about the proper API for this. I can think of the following instead of using the enum:

- `setOpenOnHover(true)` / `isOpenOnHover()`
- `setOpenOnFocus(true)` / `isOpenOnFocus()`
- `setOpenOnClick(true)` / `isOpenOnClick()`

This would be aligned with the following APIs:

- [`ContextMenu#setOpenOnClick()`](https://github.com/vaadin/flow-components/blob/60244a7113d203871efe5da6735927ce370d9512/vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java#L175)
- [MenuBar#setOpenOnHover()](https://github.com/vaadin/flow-components/blob/60244a7113d203871efe5da6735927ce370d9512/vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java#L352)

Reply:
Updated the PR to use this approach. We will try to get some feedback about this during DX tests.